packages/apps/patterns/react-appkit/src/components/CrudList/CrudList.tsx

```diff
@@ -0,0 +1,226 @@
+      );
+      const targetIndex = inputsInScope.findIndex((sibling) => sibling === event.target);
+      switch (event.key) {
+        case 'Enter':
+        case 'PageDown': {
+          if (targetIndex < inputsInScope.length - 1) {
+            event.preventDefault();
+            inputsInScope[targetIndex + 1].focus();
+          }
+          break;
+        }
+        case 'PageUp': {
+          if (targetIndex > 0) {
+            event.preventDefault();
+            inputsInScope[targetIndex - 1].focus();
+          }
+          break;
+        }
+      }
+    },
+    [hostId]
+  );
+  return { onListItemInputKeyDown, hostAttrs, itemAttrs };
+};
+
+export const CrudList = ({
+  children,
+  labelId,
+  nextItemTitle,
+  defaultNextItemTitle,
+  onNextItemTitleChange,
+  defaultItemIdOrder,
+  itemIdOrder,
+  onItemIdOrderChange,
+  slots = {}
+}: CrudListProps) => {
+  const [listItemIds, setListItemIds] = useControllableState({
+    prop: itemIdOrder,
+    defaultProp: defaultItemIdOrder,
+    onChange: onItemIdOrderChange
+  });
+  const handleDragEnd = (event: DragEndEvent) => {
+    const { active, over } = event;
+    if (active.id !== over?.id) {
+      setListItemIds((itemIds) => {
+        if (itemIds && over?.id) {
+          const oldIndex = itemIds.findIndex((id) => id === active.id);
+          const newIndex = itemIds.findIndex((id) => id === over.id);
+          return arrayMove(itemIds, oldIndex, newIndex);
+        } else {
+          return itemIds;
+        }
+      });
+    }
+  };
+  return (
+    <div role='none' {...slots.root} className={mx('contents', slots.root?.className)}>
+      <List
+        // ref={forwardedRef}
+        labelId={labelId}
+        variant='ordered-draggable'
+        selectable
+        onDragEnd={handleDragEnd}
+        listItemIds={listItemIds ?? []}
+        slots={slots.list}
+      >
+        {children}
+      </List>
+      <div className='flex'>
+        <ListItemDragHandle className='invisible' />
+        <ListItemEndcap className='invisible' />
+        <Input
+          variant='subdued'
+          label='Add new item'
+          placeholder='Add new item'
+          {...{
+            value: nextItemTitle,
+            defaultValue: defaultNextItemTitle,
+            onChange: onNextItemTitleChange
+          }}
+          slots={{
+            input: {
+              ...slots.addItem?.input,
+              className: mx('p-1 mbs-1', slots.addItem?.input?.className)
+            },
+            root: {
+              ...slots.addItem?.root,
+              className: mx('grow mbs-0', slots.addItem?.root?.className)
+            },
+            label: {
+              ...slots.addItem?.label,
+              className: mx('sr-only', slots.addItem?.label?.className)
+            }
+          }}
+        />
+        <ListItemEndcap className='invisible' />
+      </div>
+    </div>
+  );
+};
+
+export const CrudListItem = forwardRef<HTMLLIElement, CrudListItemProps>(
+  (
+    {
```

Comment:
to me it's clearer to have:

```ts
const Components = ({ a, b }) => {
  const { x, y } = b;
}
```

just easier to grok what are the params to the function when there are complex destructs.
just an opinion.

Reply:
There is not a complex destruct here, it is flat as in your first line. Maybe the `forwardRef` structure is confusing the matter?